crates/plugins/src/error.rs

```diff
@@ -0,0 +1,89 @@
+pub type PluginLookupResult<T> = std::result::Result<T, Error>;
```

Comment:
For better or worse these `Result` aliases are typically just called `Result`, e.g. `std::{fmt,io,thread}::Result`.

On the other hand its also common to e.g. `use std::io::Result as IoResult};`... :thinking:

I guess I don't actually have a strong opinion on this one, but something to be aware of.